tracer/mask_ignored_keys.go

```diff
@@ -0,0 +1,73 @@
+package tracer
+
+import (
+	"encoding/json"
+	"reflect"
+
+	"github.com/epsagon/epsagon-go/protocol"
+)
+
+const maskedValue = "****"
+
+func arrayToHitMap(arr []string) map[string]bool {
+	hitMap := make(map[string]bool)
+	for _, k := range arr {
+		hitMap[k] = true
+	}
+	return hitMap
+}
+
+func maskNestedJSONKeys(decodedJSON interface{}, ignoredKeysMap map[string]bool) interface{} {
```

Comment:
I would change this function to return also if any value has changed (so will return interface{}, bool) - so we will change values (and the final metadata value) only if needed

Reply:
how does it matter? it seems more complicated this way (adding many tests for changes across the code) and adds no benefit as assigning values takes negligible amount of time over testing and branching..

Reply:
you are right its makes the code more complicated (thats why I was wondering about it)
Do you think it still worth as we json.Marshal every time (per event/metadata key)? not sure its its negligible
if so then lets keep it that way

Reply:
It doesn't but it takes less time to add this then to explain

Reply:
I have added this to the PR